rllib/utils/spaces/space_utils.py

```diff
@@ -307,9 +307,12 @@ def convert_element_to_space_type(element: Any, sampled_element: Any) -> Any:
     def map_(elem, s):
         if isinstance(s, np.ndarray):
             if not isinstance(elem, np.ndarray):
-                raise ValueError(
-                    "Element should be of type np.ndarray but is instead of \
-                        type {}".format(type(elem)))
+                if s.shape == ():
```

Comment:
Hold on `s` here is in fact a numpy array (see if clause one line above the other one):
```
        if isinstance(s, np.ndarray):  # <- `s` is numpy, e.g. np.array(5.5).
            if not isinstance(elem, np.ndarray):  # <- but `elem` isn't, e.g. float(5.5).
                if s.shape == ():
```

Reply:
Yeah, `elem` should NOT be a torch tensor. Happy to add a check for this. ...

Reply:
Normally, we call this function with `elem` being a sample from a `gym.spaces.Space`, so numpy. But in the RecSim env case, this may be a python float. That's why I added this additional option here.

Reply:
done